osu.Game/Screens/OnlinePlay/Multiplayer/Match/Playlist/MultiplayerQueueList.cs

```diff
@@ -64,7 +64,32 @@ protected override void LoadComplete()
             {
                 base.LoadComplete();
 
-                RequestDeletion = item => multiplayerClient.RemovePlaylistItem(item.ID).FireAndForget();
+                RequestDeletion = item =>
+                {
+                    // If the requested item is the currently selected one, we proceed to a swap because we can't delete the current item
+                    if (IsSelectedItem && multiplayerClient.Room?.Playlist.Count > 1)
+                    {
+                        // We delete the second item which we cache first then we update the first one with the cache
+                        var secondItem = multiplayerClient.Room.Playlist[1];
+                        var replacingItem = new MultiplayerPlaylistItem
+                        {
+                            ID = item.ID,
+                            BeatmapID = secondItem.BeatmapID,
+                            BeatmapChecksum = secondItem.BeatmapChecksum,
+                            RulesetID = secondItem.RulesetID,
+                            RequiredMods = secondItem.RequiredMods,
+                            AllowedMods = secondItem.AllowedMods
+                        };
+
+                        multiplayerClient.RemovePlaylistItem(secondItem.ID).FireAndForget(onSuccess: () =>
+                        {
+                            multiplayerClient.EditPlaylistItem(replacingItem).FireAndForget();
+                        });
```

Comment:
To illustrate why this is wrong: think what happens if the removal of the second item succeeds but the nested edit of the first one doesn't, for whatever reason.